awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,193 @@
+"""Amazon Lake Formation Module gathering all read functions."""
+import concurrent.futures
+import itertools
+import logging
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+from pyarrow import NativeFile, RecordBatchStreamReader
+
+from awswrangler import _utils, exceptions
+from awswrangler._config import apply_configs
+from awswrangler.lakeformation._utils import wait_query
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _execute_query(
+    query_id: str,
+    token_work_unit: Tuple[str, int],
+    boto3_session: Optional[boto3.Session] = None,
+) -> pd.DataFrame:
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+    token, work_unit = token_work_unit
+    messages: NativeFile = client_lakeformation.execute(QueryId=query_id, Token=token, WorkUnitId=work_unit)["Messages"]
```

Comment:
`execute` returns Arrow streams. The issue is that we don't have control over the number nor the size of these streams. Moreover, the function returns empty streams for some specific SQL statements (e.g. `sql="SELECT * FROM gov_people WHERE gender='female' and family_name='Sanchez';"`) and `ORDER BY` does not seem to be honoured

Reply:
Nop, not even within the same work_unit scope